### DISCUSSION TITLE
Clean-up of code & tests improvements

### DISCUSSION THREAD

**Discussion by 42atomys** [bwmarrin/discordgo#1114] [Ideas]
Hi everyone ! 

When I Add GuildScheduledEvents  on discordgo (https://github.com/bwmarrin/discordgo/pull/1032), I have seen several errors in the code and its organization that could be harmonized and tidied up.

I will list these changes below and if accepted, I would like to convert this into a milestone grouping several PRs to make these changes: 

- [ ] In `restapi.go` uri and url is used without specific logic and maybe can be cleaned to `url`, following the URL convention (https://github.com/bwmarrin/discordgo/pull/1032#discussion_r815163194, thanks to @FedorLap2006)
- [x] In `structs.go`, we can think only contain struct, include custom struct helper and Marshal/Unmarshal function, this can be good to move functions on separated file (ex: `marshal.go`) 
- [ ] A file called `types.go` contains only `error`, maybe rename it `error.go`
- [ ] Add a full test suit of package (Yes I think I have a mental problem with wanting to do this)
- [ ] Write a full test suite with a discord server used only for run tests on it to perform real-life end-to-end utilisation of discordgo

Any others changes, any discussions about it are welcome !

**Comment by FedorLap2006** [comment]
Not sure about `structs` and `types`. Additionally, I was planning to do a pr with moving stuff around to make it look better. 
But it would be probably included only in `v0.24.1`

**Reply by 42atomys** [reply]
Great, its can also a side milestone or include on 0.25 its not a high priority in itself

**Comment by 42atomys** [comment]
In addition of my initial proposition : 

- [ ] Write a full test suite with a discord server used only for run tests on it to perform real-life end-to-end utilisation of discordgo
